### PR TITLE
Fix: Contacts sometimes not loading

### PR DESCRIPTION
**Pull request by kaladivo**
## Summary by CodeRabbit

* **New Features**
  * Added robust contact mapping that converts raw device contacts into domain contacts and tracks malformed contacts and phone numbers.

* **Bug Fixes**
  * Loading screen now advances to app content even if contact normalization fails.
  * Loading overlay is always hidden after contact import, even on errors.
  * Improved error reporting, permission handling, and cleanup during contact load/normalization.

* **Tests**
  * Added unit tests covering contact mapping and edge cases.

### CODE REVIEW

**Comment by kaladivo** [comment]
@greptile review

File: apps/mobile/src/state/contacts/loadContactsFromDeviceActionAtomInAppLoadingTask.ts [hunk 1]
```diff
@@ -24,10 +27,25 @@ export const loadContactsFromDeviceActionAtomInAppLoadingTaskId =
         store.set(loadingContactsFromDeviceAtom, true)
 
         yield* _(
-          store.set(loadContactsFromDeviceActionAtom),
-          Effect.catchAll(() => Effect.succeed('success' as const))
-        )
+          store.set(loadContactsFromDeviceActionAtom).pipe(
+            Effect.catchAll((error) => {
+              if (error._tag === 'UnknownContactsError') {
+                return Effect.succeed('success')
```
[the remark this answers is not in the document]
Comment (kaladivo) [reply apps/mobile/src/state/contacts/loadContactsFromDeviceActionAtomInAppLoadingTask.ts:33]: This is not true in effect error has a value!

[the remark this answers is not in the document]
Comment (kaladivo) [reply apps/mobile/src/state/contacts/loadContactsFromDeviceActionAtomInAppLoadingTask.ts:33]: Also errors like these would be caught by linter!
